dll/win32/shell32/shlfileop.cpp

```diff
@@ -1236,12 +1264,20 @@ static void copy_dir_to_dir(FILE_OPERATION *op, const FILE_ENTRY *feFrom, LPCWST
 
     if (!(op->req->fFlags & FOF_NOCONFIRMATION) && PathFileExistsW(szTo))
     {
-        if (!SHELL_ConfirmDialogW(op->req->hwnd, ASK_OVERWRITE_FOLDER, feFrom->szFilename, op))
+        CString *newPath;
+        if (lstrcmp(feFrom->szDirectory, szDestPath) == 0 && (newPath = try_find_new_name(szTo)) != NULL)
```

Comment:
... and here you can also use a stack variable instead of a pointer (as I mentioned earlier) and just check for emptiness.